src/controllers/dimension-processor.ts

```diff
@@ -0,0 +1,138 @@
+import { DimensionCreationDTO } from '../dtos/dimension-creation-dto';
+import { Dataset } from '../entities/dataset';
+import { Dimension } from '../entities/dimension';
+import { DimensionInfo } from '../entities/dimension-info';
+import { DimensionType } from '../enums/dimension-type';
+import { Revision } from '../entities/revision';
+import { Source } from '../entities/source';
+import { SourceType } from '../enums/source-type';
+import { i18next, t } from '../middleware/translation';
+import { SourceAction } from '../enums/source-action';
+
+export interface ValidatedDimensionCreationRequest {
+    datavalues: DimensionCreationDTO | null;
+    footnotes: DimensionCreationDTO | null;
+    dimensions: DimensionCreationDTO[];
+    ignore: DimensionCreationDTO[];
+}
```

Comment:
We should probably extract this to it's own file.

If `datavalues` and `footnotes` are optional then I'd avoid using `null` and just mark the props with a `?`.

Reply:
So its only used here and only used for processing the request from the client internally.  We sort the request in to various parts as each is processed slightly differently and I suspect there might be some evolution here later.  I don't expect this interface to be used outside of the dimension creation route and dimension processor so I think it makes sense to keep it with the thing that chiefly uses it.

Reply:
As for datavalues and footnotes being optional... I don't think they really are but we haven't had any thing in the specifications which have been given to us where we can insist on these things.  Footnotes might be optional longer term but datavalues are almost certainly going to be required.  This is an area for iteration later I think.